src/coder.cpp

```diff
@@ -253,13 +230,17 @@ int main(int argc, char * argv[]) {
     std::ifstream file_in;
     std::ofstream file_out;
     std::string fname;
+    size_t blocksize = DATA_BLOCK_LEN;
     int ret;
 
-    if (argc != 2 || ! strcmp(argv[1], "-h"))
-        return print_help(argv[0]);
+    if (argc < 2 || ! strcmp(argv[1], "-h"))
+        return print_help(argv[0], blocksize);
 
     fname = argv[1];
-
+    if (argc > 2)
+        blocksize = atoi(argv[2]);
```

Comment:
Maybe a check for an invalid `int`?

Reply:
https://github.com/FS-make-simple/rs-coder/blob/7b42c5c05174214929361dcfbfdfbf7ec737bf4a/src/coder.cpp#L242

Reply:
OK. I also missed the message bellow that echos configuration to the user. LGTM in that case :+1:

Reply:
Ask. Set the upper limit?

Reply:
The Upper and also the lower limit. I would also expect the application to exit with non-zero exit code rather than forcing defaults on invalid arguments.

Reply:
Maybe?:
```c
blocksize = (blocksize > data.size())? data.size() : blocksize;
```

Reply:
Or fixed upper?:
```c
#define UPPER_BLOCK_LEN 4096
blocksize = (blocksize > UPPER_BLOCK_LEN) ? UPPER_BLOCK_LEN : blocksize;
```

Reply:
>rather than forcing defaults on invalid arguments.

Silence removed:

https://github.com/FS-make-simple/rs-coder/blob/7b42c5c05174214929361dcfbfdfbf7ec737bf4a/src/coder.cpp#L266-L269

Reply:
> Maybe?:
> 
> ```c
> blocksize = (blocksize > data.size())? data.size() : blocksize;
> ```

I would prefer this one over the fixed size.

Reply:
Or interrupt? With the same code? What message? (I have a bad English: https://translate.google.ru)

Reply:
>I would prefer this one over the fixed size.

Ok.

Reply:
>Or interrupt?

`RET_ERR_ARGS`?

Reply:
> > Or interrupt?
> 
> `RET_ERR_ARGS`?

Unix/Linux programs generally use 2 for command line syntax errors.

> Or interrupt? With the same code? What message? (I have a bad English: https://translate.google.ru)

Something like `Invalid block size provided: %s.`

Reply:
Fixed all.